lib/date_range.rb

```diff
@@ -0,0 +1,38 @@
+require 'date'
+
+module Hotel
+  class DateRange
+    attr_reader :check_in, :check_out
+    
+    def initialize(check_in, check_out)
+      @check_in = check_in
+  
+      if check_out < check_in
+        raise ArgumentError, "Check-out date cannot be before check-in date"
+      else
+        @check_out = check_out
+      end
+
+      if total_stay <= 0
+        raise ArgumentError, "Invalid date range provided."
+      end
+
+    end
+
+    def total_stay
+      total_nights = (check_out - check_in).to_i
+      return total_nights
+    end
+
+    def date_check(date)
+      valid_date = (date >= @check_in && date < @check_out)
+      return valid_date
+    end
+    
+    def daterange_check(check_in, check_out)
```

Comment:
This method is also a little misleading, I would call it `overlap` and pass another `DateRange` instance to compare.  If this is the method you're using and you don't expect users to use `date_check` directly, I would also make that method private.